core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -152,6 +164,15 @@ private[spark] class CoarseMesosSchedulerBackend(
     command.build()
   }
 
+  protected def driverURL: String = {
```

Comment:
I just found it strange for this class to have protected methods, because it implies we're expecting it to be subclassed when it's not.

By the way can you make this private now that this doesn't get overridden anywhere?